device-connectors/src/testflinger_device_connectors/devices/muxpi/muxpi.py

```diff
@@ -159,8 +161,27 @@ def provision(self):
                 'the "provision_data" section of '
                 "your job_data"
             )
-        cmd = self.config.get("control_switch_local_cmd", "stm -ts")
-        self._run_control(cmd)
+        if "media" not in self.job_data["provision_data"]:
+            media = None
+            self.test_device = self.config["test_device"]
+            cmd = self.config.get("control_switch_local_cmd", "stm -ts")
+            self._run_control(cmd)
+        else:
+            media = self.job_data["provision_data"]["media"]
+            if media == "sd":
+                self.test_device = "/dev/sd-disk"
+            elif media == "usb":
+                self.test_device = "/dev/tc-disk"
+            else:
+                self.test_device = self.config["test_device"]
+            # If media option is provided, then DUT is probably capable of
+            # booting from different media, we should switch both of them to TS
+            # side regardless of which one was previously used
+            cmd = "zapper sdwire set ZAPPER; zapper typecmux set TS"
```

Comment:
A couple of things I just noticed about this section:
1. calling two commands there with ; in the middle might result in something unexpected. If the last one fails, then the whole thing will fail. But if the first one fails, it will pass. You can use && instead I think to ensure that either command failing will fail properly
2. Do you really want to catch any exception here and ignore it? It seems like if the command to set up zapper or sdwire for provisioning failed, that that should probably be a fatal error right?

Reply:
> A couple of things I just noticed about this section:
> 
> 1. calling two commands there with ; in the middle might result in something unexpected. If the last one fails, then the whole thing will fail. But if the first one fails, it will pass. You can use && instead I think to ensure that either command failing will fail properly
> 2. Do you really want to catch any exception here and ignore it? It seems like if the command to set up zapper or sdwire for provisioning failed, that that should probably be a fatal error right?

It's intentional indeed. This change can be useful even if the DUT's zapper is only connected to one of the 2 addons. Using the new parameter, you don't have to request and get a TF config change to replace the default stm commands, you can also easily switch from one to the other later on (that's typically what we did, switching from sd card to USB). So one of the 2 commands can fail legitimately. Even if the 2 of them fail, then the related /dev/xx-disk will not be detected, so the whole thing will fail anyway.